ntbs-service/Program.cs

```diff
@@ -1,152 +1,837 @@
+        app.UseHsts();
+    }
+
+    app.UseStaticFiles();
+
+    if (!app.Environment.IsEnvironment("CI"))
+    {
+        /*
+        Making this conditional is the result of serilog not playing nicely with WebApplicationFactory
+        used by the ui tests, see: https://github.com/serilog/serilog-aspnetcore/issues/105
+        Using env directly as check is an unsatisfying solution, but configuration values were not picked up consistently correctly here.
+        */
+        app.UseSerilogRequestLogging(options => // Needs to be before MVC handlers
+        {
+            options.MessageTemplate =
+                "HTTP {RequestMethod} {RequestPath} responded {StatusCode} in {Elapsed:0.0000} ms ({RequestId})";
+            // 400s get thrown e.g. on antiforgery token validation failures. In those cases we don't have
+            // an exception logged in Sentry, so we want to log at Warning level to make sure we are able to
+            // identify and cure false positives.
+            // Otherwise setting to Information to prevent duplicated exceptions in sentry.
+            options.GetLevel = (context, _, __) => context.Response.StatusCode == StatusCodes.Status400BadRequest
+                ? LogEventLevel.Warning
+                : LogEventLevel.Information;
+        });
+    }
+
+    app.UseRouting();
+
+    app.UseAuthentication();
+    app.UseAuthorization();
+    app.UseCookiePolicy();
+    app.UseSession();
+
+    app.UseMiddleware<MsOfficeLinkPrefetchMiddleware>();
+
+    if (!app.Environment.IsEnvironment("CI"))
+    {
+        app.UseMiddleware<ActivityDetectionMiddleware>();
+    }
+
+    if (app.Configuration.GetValue<bool>(Constants.AuditEnabledConfigValue))
+    {
+        app.UseMiddleware<AuditGetRequestMiddleWare>();
+    }
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapRazorPages();
+        endpoints.MapControllers();
+    });
+
+    ConfigureHangfire(app);
+
+    var cultureInfo = new CultureInfo("en-GB");
+    CultureInfo.DefaultThreadCurrentUICulture = cultureInfo;
+    CultureInfo.DefaultThreadCurrentCulture = cultureInfo;
 }
+
+void ConfigureHangfire(WebApplication app)
+{
+    if (!app.Configuration.GetValue<bool>(Constants.HangfireEnabled))
+    {
+        return;
+    }
+
+    var dashboardOptions = new DashboardOptions
+    {
+        Authorization =
+            new[]
+            {
+                new HangfireAuthorisationFilter(app.Configuration.GetSection("AdOptions")["AdminUserGroup"])
+            },
+        DisplayStorageConnectionString = false,
+    };
+    app.UseHangfireDashboard("/hangfire", dashboardOptions);
+    GlobalJobFilters.Filters.Add(new AutomaticRetryAttribute { Attempts = 0 });
+
+    var scheduledJobConfig = new ScheduledJobsConfig();
+    app.Configuration.GetSection(Constants.ScheduledJobsConfig).Bind(scheduledJobConfig);
+    HangfireJobScheduler.ScheduleRecurringJobs(scheduledJobConfig);
+}
+
+public partial class Program { }
```

Comment:
I think the only use of this class is as a marker to the entrypoint assembly. We could get rid of it and use any other class in the assembly, but I think it is quite nice having a specific class for the purpose. However, I think we should:

- Add a comment explaining the purpose
- Don't make it a partial class, as there's no point

Incidentally, part 6 of the article that I linked to earlier goes into some depth about how our integration tests still work - it's quite interesting, and well worth a read (I haven't gone through it in detail, but I scanned over it).

Reply:
If we remove `partial` here it complains that there is already a class called Program (under the hood)

Reply:
Oh in that case we should give it a different name

Reply:
And I guess put it in a different file as well

Reply:
Made EntryPoint class. There wasn't an obvious place to put it so I've put it in Helpers for now. It seemed a bit much to create a new folder for it to live in